packages/broker/src/plugins/storage/Storage.ts

```diff
@@ -111,22 +111,22 @@ export class Storage extends EventEmitter {
             prepare: true, fetchSize: 1
         }
 
-        const resultStream = this._createResultStream()
+        const resultStream = this.createResultStream()
 
-        const makeLastQuery = (bucketIds: BucketId[]) => {
-            const params = [streamId, partition, bucketIds, limit]
-
-            this.cassandraClient.execute(GET_LAST_N_MESSAGES, params, {
-                prepare: true,
-                fetchSize: 0 // disable paging
-            }).then((resultSet: Todo) => {
+        const makeLastQuery = async (bucketIds: BucketId[]) => {
+            try {
+                const params = [streamId, partition, bucketIds, limit]
+                const resultSet = await this.cassandraClient.execute(GET_LAST_N_MESSAGES, params, {
+                    prepare: true,
+                    fetchSize: 0 // disable paging
+                })
                 resultSet.rows.reverse().forEach((r: Todo) => {
                     resultStream.write(r)
                 })
-                resultStream.push(null)
-            }).catch((e: Todo) => {
-                resultStream.destroy(e)
-            })
+                resultStream.end()
```

Comment:
`push` is technically supposed to only be used internal to the stream.

`resultStream.push(null)` pushes null into the readable part of the stream.
`resultStream.write(r)` pushes the value into the writable
In a transform stream, data flows from `Writable` -> `transform(chunk)` -> `Readable`. 

By pushing `null` into the `Readable` side:
* the `null` skips the `transform` function
* it's possible data is still buffered on the `Writable` side. Pushing `null` to the readable side doesn't close the writable. Data from the writable will keep flowing through the `transform()`, but will error with `ERR_STREAM_PUSH_AFTER_EOF` if you don't check `if (this.readableEnded)` before calling `this.push` in the `transform()`.

This might not have broken things before, or maybe it did it's a bit subtle, but it matters now because every `resultStream` is now a `Transform` that implements the repeated "yield to event loop" thing.